content/_data/upgrades/2-319-1.adoc

```diff
@@ -0,0 +1,63 @@
+==== Built-In Node Name and Label Migration
```

Comment:
Is there a reason you just copied & pasted the separate documentation page, instead of linking it here? SSOT and all that, this will just create content drift.

Reply:
Alternatively, delete the existing page and rewrite the redirect here. But two separate pages with identical content, _especially_ content intended to be updated over time, makes no sense.

Reply:
I thought the readers of the upgrade guide would prefer a linear reading of the upgrade guide without needing to follow  a link.

I'm happy to either rewrite the redirect to this location or hyperlink to the existing page from here.  Do you have a preference?

Reply:
Asciidoc includes are still the better choice then: https://github.com/daniel-beck/jenkins.io/commit/331272323baf0a1d54881c26f68f826d42e93193

Reply:
> Asciidoc includes are still the better choice then: daniel-beck@3312723

That's an excellent suggestion.  I should have thought of asciidoc includes to reuse the content without copying.

Implemented in the most  recent commits